lib/apytypes/_apytypes.pyi

```diff
@@ -708,20 +708,59 @@ class APyFixedArray:
         :class:`bool`
         """
 
-    def transpose(self) -> APyFixedArray:
+    def transpose(self, axes: tuple | None = None) -> APyFixedArray:
         """
-        Return the transposition of the array.
+        Returns copy of array with axes transposed.
 
-        If the dimension of `self` is one, this method returns the a copy of `self`.
-        If the dimension of `self` is two, this method returns the matrix
-        transposition of `self`.
+        For a 1-D array, this returns it returns back the same array.
+        For a 2-D array, this is the standard matrix transpose.
+        For an n-D array, if axes are given, their order indicates how the
+        axes are permuted (see Examples). If axes are not provided, then
+        ``a.transpose(a).shape == a.shape[::-1]``.
 
-        Higher order transposition has not been implemented and will raise a
-        `NotImplementedException`.
+        Parameters
+        ----------
+        axes : tuple of ints, optional
+            If specified, it must be a tuple or list which contains a permutation
+            of [0,1,...,N-1] where N is the number of axes of `a`. The `i`'th axis
+            of the returned array will correspond to the axis numbered ``axes[i]``
+            of the input. If not specified, defaults to ``range(a.ndim)[::-1]``,
+            which reverses the order of the axes.
 
         Returns
         -------
-        :class:`APyFixedArray`
+        :class: `APyFixedArray`
+            `a` with its axes permuted.
+
+        Examples
+        --------
+        from ApyTypes import APyFixedArray
+        >>> a = APyFixedArray.from_float(
+        ...     [[1.0, 2.0, 3.0], [-4.0, -5.0, -6.0]],
+        ...     bits=5,
+        ...     frac_bits=0,
+        ... )
+        >>> a.to_numpy()
+        array([[ 1.,  2.,  3.],
+               [-4., -5., -6.]])
+        >>> a = a.transpose()
+        >>> a.to_numpy()
+        array([[ 1., -4.],
+               [ 2., -5.],
+               [ 3., -6.]])
+
+        >>> a = APyFixedArray.from_float(
+        ...     [1.0] * 6,
+        ...     bits=5,
+        ...     frac_bits=0,
+        ... ).reshape((1, 2, 3))
+        >>> a.transpose((1, 0, 2)).shape
+        (2, 1, 3)
+
+        >>> a.transpose((-2, -3, -1)).shape
+        (2, 1, 3)
+
+        -------
```

Comment:
I also realized that this produces a `<hline>` in the documentation. Although a problem that nanobind removes trailing spaces, it is better to move the Return section to the end (or something).

Reply:
I got doctest failures without that line (I've tried many different ways of doing newline). Do you want me to change functions with examples to end pydocs with the return?

Reply:
Yeah, there is an inherent issue with ending with an example as nanobind removes the final empty line which makes doctest think that it should return `""""` (I opened an issue with nanobind, but it was deemed to remote to fix, although I was welcome to provide a fix myself). I think that for now it may be better to move some other section there, yes. (It may even be that Sphinx reorders the sections again, but not sure.)